### PR TITLE
Feat - this adds basic support for packagist

### LINKED ISSUE #30: Packagist/PHP feeds
I don't really know anything about PHP, but we should add a feed for Packagist/Composer: https://packagist.org/

### PR DESCRIPTION
**Pull request by phumberdroz**
Hey,

Here is a very simple implementation for PHP feeds.

Couple of questions before I complete it:
- I would like to add tests and mock the http requests with: https://github.com/h2non/gock but did not wanna just include it. How do you feel about adding this kind of dependency?
- Currently dev updates are included and those are created when someone pushes a commit into a branch but does not add a tag. So there are versions like dev-<branch-name> (e.g. dev-master, dev-feat-packagist) should those be included? If not they can be easily filtered out.
- how should I document why I choose the api over the rss feed? (I did it because it is one endpoint for newly created packages and updates instead of 2 different rss feeds) 

closes #30

### CODE REVIEW

**Comment by jordan-wright** [comment]
Hey @phumberdroz!

Thank you so much for sending this in- very exciting! Let me know when you're ready for a full review and I'll be happy to take a look.

> I would like to add tests and mock the http requests with: https://github.com/h2non/gock but did not wanna just include it. How do you feel about adding this kind of dependency?

I'd prefer to hold off on adding dependencies for testing unless they're critical. This particular case looks like a pretty straightforward case where you can use the `httptest` package to spin up a local server to send mocked out responses. You could just move the base URL out to a global unexported variable and then override that in your tests. Let me know if you'd like a more in-depth example!

> Currently dev updates are included and those are created when someone pushes a commit into a branch but does not add a tag. So there are versions like dev- (e.g. dev-master, dev-feat-packagist) should those be included? If not they can be easily filtered out.

I think we can start by including them and filter out if we find they're causing too much noise.

> how should I document why I choose the api over the rss feed? (I did it because it is one endpoint for newly created packages and updates instead of 2 different rss feeds)

I think documentation in general can be done using Go's standard documentation syntax. You can either include it in the documentation for your `Latest` function or, maybe better, add a `doc.go` that just includes the standard:

```go
// package packagist fetches packages from etc. etc
// more info here about how you picked your design
package packagist
```

File: feeds/packagist/packagist_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
...lyt/free-to-game-api-wrapper","description":"Wrapper for API https://www.freetogame.com","keywords":["api-wrapper","free to game"],"homepage":"","version":"dev-master","version_normalized":"dev-master","license":["MIT"],"authors":[{"name":"Niall Tiernan","email":"nialltiernan93@gmail.com"}],"source":{"type":"git","url":"https://github.com/nialltiernan/freetogame-api-wrapper.git","reference":"c3afaa087afb42bfaf40fc3cda1252cd9a653d7f"},"dist":{"type":"zip","url":"https://api.github.com/repos/nialltiernan/freetogame-api-wrapper/zipball/c3afaa087afb42bfaf40fc3cda1252cd9a653d7f","reference":"c3afaa087afb42bfaf40fc3cda1252cd9a653d7f","shasum":""},"type":"library","time":"2021-02-28T12:20:03+00:00","autoload":{"psr-4":{"FreeToGame\\":"src/"}},"default-branch":true,"require":{"php":"^8.0","guzzlehttp/guzzle":"^7.2","lukasoppermann/http-status":"^3.1"},"require-dev":{"codeception/codeception":"^4.1","codeception/module-phpbrowser":"^1.0.0","codeception/module-asserts":"^1.0.0","hoa/console":"^3.17"},"support":{"issues":"https://github.com/nialltiernan/freetogame-api-wrapper/issues","source":"https://github.com/nialltiernan/freetogame-api-wrapper/tree/v1.0.0"}}]},"minified":"composer/2.0"}`,
```
Comment (phumberdroz) [inline feeds/packagist/packagist_test.go:55]: Let me know if I should exchange this with fake data.

**Comment by phumberdroz** [comment]
I think this can be reviewed now.

**Review by jordan-wright — CHANGES_REQUESTED** [review]
This is looking great!

I've added some comments but they're mostly just stylistic. The approach looks just fine to me.

Thanks so much for sending this in!

File: feeds/packagist/packagist.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"time"
+
+	"github.com/ossf/package-feeds/feeds"
+)
+
+const FeedName = "packagist"
+
+var (
+	UpdateHost  = "https://packagist.org"
+	VersionHost = "https://repo.packagist.org"
```
[the remark this answers is not in the document]
Comment (phumberdroz) [reply feeds/packagist/packagist.go]: I usually write test from the outside with `package packagist_test` which results in the test file only having access to exported functions/vars etc. 

I changed it. 👍

**Review by jordan-wright — CHANGES_REQUESTED** [review]
Apologies for the delay in getting back to you! This looks good. The only change I'd request is that we change `packagistversion.go` -> `version.go` and the current `version.go` -> `packagist.go` to match the other feeds.

When that's in good shape, feel free to tag me and I'll get this merged in. Thank you for all the great work here and for all your patience!

**Comment by phumberdroz** [comment]
hey @jordan-wright,

I renamed the files as you requested.

And no worries this is an important effort which can help a lot of people to stay away from trouble.

**Pull request merged**